### PR TITLE
Update animal table

### PR DESCRIPTION
**Pull request by Yemisirach**
- [ ] Add a column species of type string to the animals table and modify [schema.sql](https://github.com/Yemisirach/Vet_clinic_database/blob/Update-Animals-Table/schema.sql) file.
- [ ] Insert new data in the table and add these insertions to the [data.sql](https://github.com/Yemisirach/Vet_clinic_database/blob/Update-Animals-Table/queries.sql) file.

- [ 👍  ] Inside a transaction update the animals table and roll back the change:
Screenshot1
Screenshot2
Screenshot3
Screenshot4
Screenshot5
Screenshotnew

- [ ] Transaction 1

Screenshot6
Screenshot7

- [ ] Transaction 2

Screenshot8trans2
Screenshot10trance

- [ ] Transaction 3

Screenshot13tran

 - [x] How many animals are there, How many animals have never tried to escape,, and What is the average weight of animals

Screenshot11tran
- [ ]  Who escapes the most, neutered or not neutered animals, What is the minimum and maximum weight of each type of animal, and What is the average number of escape attempts per animal type of those born between 1990 and 2000
Screenshot12tran

- [ ] Add these queries to your [queries.sql](https://github.com/Yemisirach/Vet_clinic_database/blob/Update-Animals-Table/queries.sql) file.

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -8,3 +8,32 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
...
 SELECT * FROM animals WHERE name != 'Gabumon';
 SELECT * FROM animals WHERE weight_kg BETWEEN 10.3 AND 17.4;
+
+
+/* Day 2 */
+
+/* Transaction 1 */
+
+BEGIN;
+UPDATE animals SET species = 'unspecified';
```
Comment (t-yanick) [inline queries.sql]: - [ ] Great job working on this milestone. After the command of line 18, you are supposed to have a command on line 19 before the `ROLLBACK` to verify that the change was made.
See [Project requirement](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/vet_clinic_query_update_tables.md) below:
**Inside a transaction update the animals table by setting the species column to unspecified.` Verify that change was made. `Then roll back the change and verify that the species columns went back to the state before the transaction.**

File: queries.sql [hunk 2]
```diff
@@ -8,3 +8,32 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
...
+ROLLBACK;
```
Comment (t-yanick) [inline queries.sql:19]: - [ ] After the ROLLBACK, you have to verify again that the change was made.

File: queries.sql [hunk 3]
```diff
@@ -8,3 +8,32 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
...
+UPDATE animals SET species = 'digimon' WHERE name ilike '%mon%';
+UPDATE animals SET species = 'pokemon' WHERE species is null;
+COMMIT;
+
```
Comment (t-yanick) [inline queries.sql:27]: - [ ] After the COMMIT, you have to verify that the change persist. Please check the [Project requirements](https://github.com/microverseinc/curriculum-databases/blob/main/db-structure/vet_clinic_query_update_tables.md) once more.

File: queries.sql [hunk 4]
```diff
@@ -8,3 +8,32 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
...
+/* Transaction 2 */
+
```
Comment (t-yanick) [inline queries.sql:31]: - [ ] You have missed one part of the project requirements which I put below. Kindly add these queries.
**Now, take a deep breath and... Inside a transaction delete all records in the animals table, then roll back the transaction.
After the rollback verify if all records in the animals table still exists. After that, you can start breathing as usual ;)**

File: schema.sql [hunk 5]
```diff
@@ -1,10 +1,12 @@
...
-    id int NOT NULL IDENTITY(1,1),
+    id              INT,
     name            CHAR(20),
     date_of_birth   DATE,
     escape_attempts INT, 
     neutered        BOOLEAN,
     weight_kg       DECIMAL
 );
+
+ALTER TABLE animals ADD species CHAR(20);
```
Comment (t-yanick) [inline schema.sql:12]: - [ ] I think you can use the VARCHAR data type which has a variable length limit rather than CHAR with fixed length

Comment (t-yanick) [inline schema.sql:12]: - [ ] I think you can use the VARCHAR data type which has a variable length limit rather than CHAR with fixed length

**Review by t-yanick — CHANGES_REQUESTED** [review]
# STATUS: CHANGES REQUIRED

Hi @Yemisirach , 

### Good job so far!
### To Highlight! :clap:  :green_circle: 

:heavy_check_mark:  SQL commands put in the correct files
:heavy_check_mark: Screenshots of queries added to PR description

There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: queries.sql (hunk 1 above)
Comment (Yemisirach) [reply queries.sql]: - [ ] Thanks for your review @t-yanick, I think that's something I've already done as shown in the description [screenshot](https://user-images.githubusercontent.com/104437323/231373132-cfc567ab-0726-43b6-89cc-74fa10b7adb2.png)
  ,  [screenshot](https://user-images.githubusercontent.com/104437323/231373242-0f2aca85-c206-45a1-a686-71b67631b8a5.png) ,  [screenshot](https://user-images.githubusercontent.com/104437323/231373308-3ec396d9-8f0a-43b8-a75e-f9d85c8c9497.png),    Thanks.

File: queries.sql (hunk 2 above)
Comment (Yemisirach) [reply queries.sql:19]: - [ ] Thanks for your review @t-yanick, I think that's something I've already done as shown in the description  [screenshot](https://user-images.githubusercontent.com/104437323/231374251-66ed2e6c-5bd2-43b0-ad61-e7e2a599c4b5.png).  Thanks.

**Comment by t-yanick** [comment]
- [ ] I saw the screenshots, but I think your code has to reflect what is displayed on the screenshots

File: queries.sql (hunk 2 above)
Comment (t-yanick) [reply queries.sql:19]: You're right, but I think you have to add the command in your code so that the screenshot reflects what is in the code

**Review by pinkmoon25 — APPROVED** [review]
Hi @Yemisirach  

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
Changes implemented :heavy_check_mark: 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**